account-kit/wallet-client/src/client/client.e2e.test.ts

```diff
@@ -1,9 +1,49 @@
 import { LocalAccountSigner } from "@aa-sdk/core";
 import { alchemy, arbitrumSepolia } from "@account-kit/infra";
 import { describe, expect, it } from "bun:test";
-import { createPublicClient, zeroAddress, type Address, type Hex } from "viem";
+import {
+  createPublicClient,
+  zeroAddress,
+  type Address,
+  type WaitForCallsStatusReturnType,
+} from "viem";
 import { createSmartWalletClient, type SmartWalletClient } from "./index.js";
-import { sleep } from "bun";
+import type { PrepareCallsParams } from "./actions/prepareCalls.js";
+
+// We want to test both the "unroll each step" method and the full e2e "sendCalls" method.
+const sendVariants: Array<
+  (
+    client: SmartWalletClient,
+    input: PrepareCallsParams,
+    assertCallIdsSize?: number,
+  ) => Promise<WaitForCallsStatusReturnType>
+> = [
+  // Send calls
+  async (client, input, assertCallIdsSize) => {
+    const result = await client.sendCalls(input);
+    if (assertCallIdsSize != null) {
+      expect(result.preparedCallIds).toHaveLength(assertCallIdsSize);
+    }
+    return client.waitForCallsStatus({ id: result.preparedCallIds[0] });
+  },
+  // Prepare, sign, send calls
+  async (client, input, assertCallIdsSize) => {
+    const preparedCalls = await client.prepareCalls(input);
+    const signedCalls = await client.signPreparedCalls(preparedCalls);
+    const result = await client.sendPreparedCalls({
+      ...signedCalls,
+      ...(input.capabilities?.permissions != null
+        ? { capabilities: { permissions: input.capabilities.permissions } }
+        : {}),
+    });
+
+    if (assertCallIdsSize != null) {
+      expect(result.preparedCallIds).toHaveLength(assertCallIdsSize);
```

Comment:
why does the `assertCallIdsSize` having the same length verify that `sendCalls` was successful?

Reply:
answered in response to the previous comment!